pubsub/callback/callback.go

```diff
@@ -0,0 +1,41 @@
+package callback
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/micromdm/micromdm/pubsub"
+)
+
+type CallbackPublisher struct {
+	url  string
+	next pubsub.Publisher
+}
+
+func (p *CallbackPublisher) Publish(ctx context.Context, topic string, msg []byte) error {
+
+	// send to callback url
+	var buf bytes.Buffer
+	var event = struct {
+		Topic   string `json:"topic"`
+		Message []byte `json:"message"`
+	}{
+		Topic:   topic,
+		Message: msg,
```

Comment:
For a byte array being JSON'd that'll turn it into base64 data. I'd want to avoid that for the command callback. The thought there is you should only need an HTTP server and a JSON parser for the callback (which may or may not decode base64 for you, so best to avoid it) just as you only need a JSON encoder and an HTTP client to _submit_ commands.

Also most (all?) of the other events on the bus are protobuf which I don't know needs to be exposed for this particular callback. They _could_ be, but that's outside the scope of what I was thinking for this callback. I.e. dirt simple JSON (only) for command responses.